tests/api/api/test_pipelines.py

```diff
@@ -296,11 +306,22 @@ def test_create_pipeline(
     with open(str(pipeline_file_path)) as file:
         contents = file.read()
     _mock_pipelines_creation(kfp_client_mock)
-    response = client.post(
-        f"projects/{project}/pipelines",
-        data=contents,
-        headers={"content-type": "application/yaml"},
-    )
+    app.dependency_overrides[deps.authenticate_request] = mock_authenticate_request
+
+    with patch(
+        target="server.api.utils.auth.verifier.AuthVerifier.get_or_create_access_key",
+        return_value="mock-access-key",
+    ):
+        with patch(
+            target="server.api.api.utils.create_secret_env_vars_for_pipeline",
+            return_value=[V1EnvVar(name="test", value="test")],
+        ):
```

Comment:
You can chain these
```suggestion
    ), patch(
            target="server.api.api.utils.create_secret_env_vars_for_pipeline",
            return_value=[V1EnvVar(name="test", value="test")],
        ):
```

Reply:
actually don't mock this - you can use `k8s_secrets_mock` and then you can also assert that the secret was created as expected